lib/count_words.rb

```diff
@@ -0,0 +1,62 @@
+class CountWords
+
+    # parse text
+    def run(filename)
+
+        # read the text file
+        text = readFile(filename)
+
+        # strip out non alpha characters
+        cleanedText = stripCharacters(text.downcase)
+
+        # count all words
+        wordCount = countAllWords(cleanedText)
+
+        # reverse sort word counts
+        sortedWords = reverseSortWordCounts(wordCount)
+
+        # display word plus it's found value
+        displayWordCounts(sortedWords)
+    end
+
+    # get the contents of a file
+    def readFile(filename)
+        # TODO 1. ensure file extension is .txt
+        #      2. limit file size or refactor app to stream lines
+        #      3. ensure the file exists
+        File.read(filename)
+    end
+
+    # strip non alpha characters from text
+    def stripCharacters(text)
+        text.scan(/[\w']+/)
+    end
+
+    # count all the words and add them to a hash
+    def countAllWords(cleanedText)
+        wordCount = {}
```

Comment:
anytime you initialize a collection to then iteratively build it up is a clue to maybe use a method like `.map` or `.reduce` which are common in a lot of languages.

here i'd recommend using a method like `.reduce` or `.each_with_object`.   
lines 37-46 could also be written:

```ruby
cleaned_text.reduce(Hash.new(0)) { |hash,word| hash[word] += 1; hash }
```

### or 

```ruby
cleaned_text.each_with_object(Hash.new(0)) { |word, hash| hash[word] += 1}
```